unittests/producer_schedule_savanna_tests.cpp

```diff
@@ -434,4 +434,36 @@ BOOST_AUTO_TEST_CASE( switch_producers_test ) try {
 
 } FC_LOG_AND_RETHROW()
 
+// This is to verify the bug reported by https://github.com/AntelopeIO/spring/issues/454
+// is fixed.
+BOOST_FIXTURE_TEST_CASE( policy_switching_corner_case_test, validating_tester ) try {
+   // With regular validating_tester, we have already transitioned into Savanna
+
+   // In round 1, a block proposes a proposer policy.
+   create_accounts( {"alice"_n} );
+   set_producers( {"alice"_n} );
+   auto b = produce_block();
+   auto index = b->timestamp.slot % config::producer_repetitions; // current index in current round
+   produce_blocks(config::producer_repetitions - index - 1); // to end of the round
+
+   // In round 2, the block in the last time slot of the round is not present.
+   produce_blocks(config::producer_repetitions - 1);
+   const auto time_to_skip = fc::milliseconds(config::block_interval_ms);
```

Comment:
Default `time_to_skip` is `fc::microseconds default_skip_time = fc::milliseconds(config::block_interval_ms)` so this is just doing the default produce one block. I think you need `2 * config::block_interval_ms`.

Reply:
Tried `2 * config::block_interval_ms`. Test still passed. Will dig into it tomorrow.